src/OpenLoco/Windows/Construction/OverheadTab.cpp

```diff
@@ -121,26 +123,224 @@ namespace OpenLoco::Ui::Windows::Construction::Overhead
+        _1135F52 = viewport;
+
+        if (interaction.type != ViewportInteraction::InteractionItem::track)
+        {
+            return std::nullopt;
+        }
+
+        auto* elTrack = reinterpret_cast<Map::TileElement*>(interaction.object)->asTrack();
+        if (elTrack == nullptr)
+        {
+            return std::nullopt;
+        }
+
+        GameCommands::TrackModsPlacementArgs args;
+        args.type = _lastSelectedMods;
+        args.pos = Map::Pos3(interaction.pos.x, interaction.pos.y, elTrack->baseZ() * 4);
+        args.rotation = elTrack->unkDirection();
+        args.trackId = elTrack->trackId();
+        args.index = elTrack->sequenceIndex();
+        args.trackObjType = elTrack->trackObjectId();
+        args.modSection = _lastSelectedTrackModSection;
+        return { args };
+    }
+
+    static uint32_t placeRoadModGhost(const GameCommands::RoadModsPlacementArgs& args)
+    {
+        auto res = GameCommands::do_40(GameCommands::Flags::apply | GameCommands::Flags::flag_1 | GameCommands::Flags::flag_3 | GameCommands::Flags::flag_5 | GameCommands::Flags::flag_6, args);
+        if (res != GameCommands::FAILURE)
+        {
+            _byte_522096 = _byte_522096 | (1 << 4);
+            _modGhostPos = args.pos;
+            _modGhostRotation = args.rotation;
+            _modGhostTrackId = args.roadId;
+            _modGhostTileIndex = args.index;
+            _modGhostTrackObjId = args.roadObjType | (1 << 7); // This looks wrong!
+        }
+        return res;
+    }
+
+    static uint32_t placeTrackModGhost(const GameCommands::TrackModsPlacementArgs& args)
+    {
+        auto res = GameCommands::do_17(GameCommands::Flags::apply | GameCommands::Flags::flag_1 | GameCommands::Flags::flag_3 | GameCommands::Flags::flag_5 | GameCommands::Flags::flag_6, args);
+        if (res != GameCommands::FAILURE)
+        {
+            _byte_522096 = _byte_522096 | (1 << 4);
+            _modGhostPos = args.pos;
+            _modGhostRotation = args.rotation;
+            _modGhostTrackId = args.trackId;
+            _modGhostTileIndex = args.index;
+            _modGhostTrackObjId = args.trackObjType;
+        }
+        return res;
+    }
+
     // 0x0049EC15
     static void onToolUpdate(Window& self, const WidgetIndex_t widgetIndex, const int16_t x, const int16_t y)
     {
-        registers regs;
-        regs.esi = X86Pointer(&self);
-        regs.dx = widgetIndex;
-        regs.ax = x;
-        regs.bx = y;
-        call(0x0049EC15, regs);
+        if (widgetIndex != widx::image)
+        {
+            return;
+        }
+
+        if (_trackType & (1 << 7))
+        {
+            auto placementArgs = getRoadModsPlacementArgsFromCursor(x, y);
+            if (!placementArgs || ((placementArgs->roadObjType | (1 << 7)) != _trackType))
+            {
+                removeConstructionGhosts();
+                if (0x80000000 != _modCost)
```

Comment:
Any particular reason for using [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)?

Reply:
Just cause i copy pasted from a different line and only needed to modify the left hand